src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -402,12 +385,21 @@ void OTARequestor::ApplyUpdate()
     ConnectToProvider(kApplyUpdate);
 }
 
-void OTARequestor::NotifyUpdateApplied()
+void OTARequestor::NotifyUpdateApplied(uint32_t version)
 {
+    // New version is executing so update where applicable
+    VerifyOrReturn(Basic::Attributes::SoftwareVersion::Set(kRootEndpointId, version) == EMBER_ZCL_STATUS_SUCCESS);
```

Comment:
@Damian-Nordic This kind of continues on our last convo on this topic. The idea is the caller (be it the application or in Linux case, the dummy ImageProcessor impl) will be the one that knows the new version that has booted and will call into OTA Requestor core to inform of this update and it'll be the core's job to update it's internal version as well as the Basic cluster attribute.